### DISCUSSION TITLE
Two different Shapes for same Element in stencil & Canvas

### DISCUSSION THREAD

**Discussion by saravntbe** [clientIO/joint#2019] [Q&A]
- can we have 1 shape rendered in the stencil & other shape been rendered when the same element is dropped in the canvas.  Right now we have two shapes for an element & hide the unwanted shape using css based on stencil & canvas.
- is it possible to have custom drag handle for an element when it's inside the stencil, so that user should only be able to drag that element using that handle to drop on canvas.

**Comment by kumilingus** [comment]
1. Here's a [demo](https://www.jointjs.com/demos/stencil-vs-diagram-elements).
2. You can set `pointerEvents: 'none'` on any part of the shape which you don't want to be draggable.
